src/igraph_eulerian.c

```diff
@@ -415,11 +415,78 @@ igraph_integer_t igraph_eulerian_path_directed(igraph_t *graph, igraph_vector_t
 
     return IGRAPH_SUCCESS;
 }
+/*
+igraph_integer_t igraph_eulerian_path(igraph_t *graph, igraph_vector_t *res) {
+    igraph_bool_t has_path;
+    igraph_bool_t cycle;
 
-igraph_integer_t igraph_eulerian_paths(igraph_t *graph, igraph_vector_t *res) {
+    igraph_is_eulerian(graph, &has_path, &cycle);
+
+    if (has_path == 0 && has_cycle == 0) {
+        IGRAPH_WARNING("Euler cycle not possible");
+        return IGRAPH_FAILURE;
+    }
+    
+    if (igraph_is_directed(graph)) {
+        return igraph_eulerian_path_directed(graph, res);
+    } else {
+        return igraph_euler_path_undirected(graph, res);
+    } 
+}
+
+igraph_integer_t igraph_eulerian_cycle(igraph_t *graph, igraph_vector_t *res) {
+    igraph_bool_t has_path;
+    igraph_bool_t cycle;
+
+    igraph_is_eulerian(graph, &has_path, &cycle);
+
+    if (has_cycle == 0) {
+        IGRAPH_WARNING("Euler cycle not possible");
+        return IGRAPH_FAILURE;
+    }
+    
     if (igraph_is_directed(graph)) {
         return igraph_eulerian_path_directed(graph, res);
     } else {
         return igraph_euler_path_undirected(graph, res);
     } 
-}
\ No newline at end of file
+}
+*/
+
+
+igraph_integer_t igraph_eulerian_paths(igraph_t *graph, igraph_vector_t *res) {
+    double vec1, vec2;
+    igraph_integer_t curr_edge;
+    igraph_bool_t error;
+    igraph_vector_t vector_res;
+
+    error = 0;
+    IGRAPH_CHECK(igraph_vector_init(&vector_res, 0));
+    IGRAPH_FINALLY(igraph_vector_destroy, &vector_res);
+
+    if (igraph_is_directed(graph)) {
+        igraph_eulerian_path_directed(graph, &vector_res);
+
+        for (int i = 0; i < igraph_vector_size(&vector_res) - 1; i++) {
+            vec1 = VECTOR(vector_res)[i];
+            vec2 = VECTOR(vector_res)[i+1];
+
+            igraph_get_eid(graph, &curr_edge, (int) vec1, (int) vec2, 1, error);
```

Comment:
Repeated call to `igraph_get_eid` are going to be extremely slow. They must do a full search for the edge. Can you avoid using this function?

Also, you are casting to `int` when the expected argument is `igraph_integer_t`. Do not assume that these are the same. Use the same type that is required for the argument.

Reply:
Thank you for your reply. To be honest, I do not really like this implementation as well (especially when operating on especially big graphs) but at the moment, I do not really have any other way of doing this. I really wanted to do this process (getting the edge id's) as the eulerian walk process is happening, but I soon realised that deleting an edge will also change all the edge id's, so I decided not to do this. If I have a better idea, I will definitely change this.